### PR TITLE
Skip `channel_update` signature checks if we have a newer state

### PR DESCRIPTION
**Pull request by TheBlueMatt**
`channel_update` messages already have their signatures checked
with the network graph write lock held, so there's no reason to
check the signatures before doing other quicker checks first,
including checking if we're already aware of a newer update for the
channel.

This reduces common-case CPU usage as `channel_update`s are sent
rather liberally over the p2p network to gossip them.

### CODE REVIEW

File: lightning/src/routing/network_graph.rs [hunk 1]
```diff
@@ -1394,7 +1394,11 @@ impl NetworkGraph {
 						} else {
 							chan_was_enabled = false;
 						}
+					}
+				}
 
+				macro_rules! maybe_update_channel_info {
```
Comment (jkczyz) [inline lightning/src/routing/network_graph.rs:1400]: Should we drop the `maybe_` from the macro name?

File: lightning/src/routing/network_graph.rs [hunk 2]
```diff
@@ -1394,7 +1394,11 @@ impl NetworkGraph {
...
+					( $target: expr, $src_node: expr) => {
```
Comment (jkczyz) [inline lightning/src/routing/network_graph.rs:1401]: Looks like `$src_node` is not used in either macro. Probably could get rid of `$target`, too, and just have the macro evaluate to the `ChannelUpdateInfo` and assign from call sites.

File: lightning/src/routing/network_graph.rs [hunk 3]
```diff
@@ -1397,8 +1397,8 @@ impl NetworkGraph {
 					}
 				}
 
-				macro_rules! maybe_update_channel_info {
-					( $target: expr, $src_node: expr) => {
+				macro_rules! get_new_channel_info {
+					() => {
```
Comment (jkczyz) [inline lightning/src/routing/network_graph.rs:1401]: Need another set of braces.

File: lightning/src/routing/network_graph.rs [hunk 4]
```diff
@@ -1414,31 +1414,31 @@ impl NetworkGraph {
 							},
 							last_update_message
 						};
-						$target = Some(updated_channel_update_info);
+						Some(updated_channel_update_info);
```
Comment (jkczyz) [inline lightning/src/routing/network_graph.rs:1417]: Will need to drop the semi-colon. Though could make this macros a single expression, FWIW.

**Review by jkczyz — DISMISSED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without changes.

**Review by valentinewallace — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]

**Pull request merged**